pkg/shipper/shipper.go

```diff
@@ -338,16 +355,17 @@ func (s *Shipper) Sync(ctx context.Context) (uploaded int, err error) {
 	}
 
 	s.metrics.dirSyncs.Inc()
-	if uploadErrs > 0 {
-		s.metrics.uploadFailures.Add(float64(uploadErrs))
-		return uploaded, errors.Errorf("failed to sync %v blocks", uploadErrs)
+	if uploadErrs > 0 || len(failedBlocks) > 0 {
+		s.metrics.uploadFailures.Add(float64(uploadErrs + len(failedBlocks)))
```

Comment:
Should we count failed blocks as upload failures? There is no upload for those blocks. I would rather add a new metric for corrupted block meta instead of counting them as upload failures

Reply:
My idea is to reuse the metric as it has the same meaning. The block was no uploaded to obj store. Do you prefer to add a new metric and user track both?